lib/tasks/update_participant_profile_states.rake

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require "rake"
+
+namespace :update_participant_profile_states do
+  desc "Back filling cpd_lead_providers on participant_profile_states"
+  task populate_cpd_lead_provider: :environment do
+    PaperTrail.request.controller_info = {
+      reason: "Back filling cpd_lead_providers on participant_profile_states",
+    }
+
+    participant_profiles_with_a_single_induction_record = ParticipantProfile.joins(:induction_records).group(:id).having("count(*) < 2")
+
+    participant_profiles_with_a_single_induction_record.each do |pp|
```

Comment:
you want an includes here, something like `includes(:participant_profile_states)` (cant remember if `include` or `includes`). otherwise you'll cause an n+1 query

Reply:
then on line 16 you may want to also include more associations

Reply:
This is going to be a one-off rake task and then delete, but it will be a lot of records